### PR TITLE
Log when the latest dist tag has been downloaded

### PR DESCRIPTION
**Pull request by gregjopa**
This PR adds logging so we know when the `latest` dist tag has been downloaded with grabthar `poll()`.

### CODE REVIEW

File: server/components/buttons/script.js [hunk 1]
```diff
@@ -103,5 +105,9 @@ export async function getSmartPaymentButtonsClientScript({ logBuffer, cache, deb
     const { version } = await watcher.get(ACTIVE_TAG);
     const script = await watcher.read(debug ? BUTTON_CLIENT_JS : BUTTON_CLIENT_MIN_JS, ACTIVE_TAG);
 
+    watcher.get(LATEST_TAG).then(({ version: deployVersion }) => {
```
Comment (gregjopa) [inline server/components/buttons/script.js:108]: Note that we do not want this promise to be blocking. That's why it's not using `await`. We only want to fire off the log when it's done so we now the deploy version (clientsdknodeweb takes the same approach).

File: server/components/buttons/script.js [hunk 2]
```diff
@@ -103,5 +105,9 @@ export async function getSmartPaymentButtonsClientScript({ logBuffer, cache, deb
...
+        logBuffer.info(`deploy_client_version_${ deployVersion.replace(/[^0-9]+/g, '_') }`, { deployVersion });
```
Comment (bluepnume) [inline server/components/buttons/script.js:109]: Can we:

- Log the active version too
- Do this in the other places we use grabthar too, to help diagnose future issues

File: server/watchers.js [hunk 3]
```diff
@@ -10,17 +10,42 @@ import { SDK_RELEASE_MODULE, SMART_BUTTONS_MODULE, MODULE_POLL_INTERVAL, SDK_CDN
...
+        |}
+    }
+|};
+
 type Watcher = {|
-    get : (tag? : string) => Promise<{|
-        version : string
-    |}>,
+    get : (tag? : string) => Promise<ModuleDetails>,
+    getTag : () => Promise<ModuleDetails>,
```
Comment (gregjopa) [inline server/watchers.js:28]: I noticed that clientsdknodeweb returns `getTag()` and `getDeployTag()` as convenience functions. I'm thinking we can do the same here and also attach logging to it.

https://github.paypal.com/Checkout/clientsdknodeweb/blob/master/server/sdk/watcher.js#L45-L46

File: server/components/buttons/script.js (hunk 2 above)
Comment (gregjopa) [reply server/components/buttons/script.js]: @bluepnume thanks for the feedback! I think I got the logging added everywhere for both active and deploy tags. Please review again.

**Review by bluepnume — APPROVED** [review]
LGTM!

**Pull request merged**